### PR TITLE
Monitor request time

### PR DESCRIPTION
**Pull request by clxswalther**

### CODE REVIEW

File: src/sysmonitor/SystemMonitor.php [hunk 1]
```diff
@@ -48,8 +51,9 @@ public function collectStats(RequestStatsEvent $evt)
         // we won't even collect data for requests which don't hit either of these min-values, to prevent unnecessary garbage.
         $minQueries = 130;
         $minMemory = 80;
+        $minTime = 3;
 
-        if ($evt->usedQueries < $minQueries && $evt->peakMemory < $minMemory) {
+        if ($evt->usedQueries < $minQueries && $evt->peakMemory < $minMemory && $evt->requestTime < $minTime) {
```
Comment (staabm) [inline src/sysmonitor/SystemMonitor.php:56]: this change doesnt make sense. it means we will only collect request stats for event taking longer then 3 secs ...

File: src/sysmonitor/SystemMonitor.php [hunk 2]
```diff
@@ -90,6 +94,9 @@ private function rateAndStore(SystemEvent $sysEvt)
         if ($count >= 10 && $evt instanceof RequestExceptionEvent) {
             // .. based on frequency of the same failure
             $sysEvt->severity = SystemEvent::SEVERITY_URGENT;
+        } elseif ($count >= 10 && $evt instanceof RequestStatsEvent && $evt->requestTime > 3) {
```
Comment (staabm) [inline src/sysmonitor/SystemMonitor.php:97]: I dont think we need this case either (especially not with a hardcoded 3 secs threshold).

**Review by staabm — CHANGES_REQUESTED** [review]

File: src/sysmonitor/SystemMonitor.php (hunk 1 above)
Comment (clxswalther) [reply src/sysmonitor/SystemMonitor.php:56]: Sure. Any special reason why we shouldn't ignore requests that don't hit any of the limits?

File: src/sysmonitor/SystemMonitor.php (hunk 2 above)
Comment (staabm) [inline src/sysmonitor/SystemMonitor.php]: scratch that, we defined that specific case in the spec.

File: src/sysmonitor/SystemMonitor.php (hunk 1 above)
Comment (staabm) [reply src/sysmonitor/SystemMonitor.php:56]: hmm I guess you are right. it only discards if neither of the conditions is met. 👍

**Comment by staabm** [comment]
adjusted per discussion and merged. released in v0.3

thx.

**Pull request closed without merging**